src/js/views/homeView.js

```diff
@@ -171,7 +178,353 @@ export const setEditUI = (state) => {
+            document.querySelector('#dashboard-back-btn').addEventListener('click', async e => {
+                const btn2 = e.target.closest('#dashboard-back-btn');
+
+                if(btn2) {
+                    // clear right screen
+                    clearScreen();
+                    // render temp 2
+                    homeCtrl.homeScreen(state);
+                }
+            });
+        }
+    });
+
+    document.querySelector('#senior-post-opp-btn').addEventListener('click', async e => {
+        const btn = e.target.closest('.add-referral-template');
+        if(btn) {
+            document.querySelector("#senior-post-opp").classList.remove("invisible");
+
+            document.querySelector("#senior-post-opp").addEventListener('click',e => {
+                e.preventDefault();
+                const btn2 = e.target.closest('.signUp-small-container');
+                if(btn2 == null){
+                    document.querySelector("#senior-post-opp").classList.add("invisible");
+                }
+            });
+            document.querySelector("#senior-post-opp-back-btn").addEventListener('click',e => {
+                e.preventDefault();
+                document.querySelector("#senior-post-opp").classList.add("invisible");
+            });
+            document.querySelector('#senior-post-opp-post-btn').addEventListener('click', async e => {
+                state.home.seniorPostCategory = document.querySelector("#senior-post-opp-category");
+                state.home.seniorPostCompany = document.querySelector("#senior-post-opp-company");
+                state.home.seniorPostTitle = document.querySelector("#senior-post-opp-title");
+                state.home.seniorPostLimit = document.querySelector("#senior-post-opp-limit");
+                state.home.seniorPostDeadline = document.querySelector("#senior-post-opp-deadline");
+                state.home.seniorPostDescription = document.querySelector("#senior-post-opp-description");
+            
+
+                state.home.seniorPostNewOpp(state);
+                document.querySelector("#senior-post-opp").classList.add("invisible");
+                clearScreen();
+                await homeCtrl.homeScreen(state);
+
+                state.home.seniorPostCategory.value = "";
+                state.home.seniorPostCompany.value = "";
+                state.home.seniorPostTitle.value = "";
+                state.home.seniorPostLimit.value = "";
+                state.home.seniorPostDeadline.value = "";
+                state.home.seniorPostDescription.value = "";
+
+                console.log("Senior New Opp Posted");
+            });
+        }
+    });
+};
+
+let seniorOppDetailCounter = [0,0,0,0];
+
+async function asyncForEach(array, callback) {
+    for (let index = 0; index < array.length; index++) {
+        await callback(array[index], index, array);
+    }
+}
+
+const updateSeniorOppDetails = function(state,list,info){

```

Comment:
What is `list`? Be more descriptive please and/or add documentation

Reply:
Done